src/mongo/db/storage/wiredtiger/wiredtiger_kv_engine.cpp

```diff
@@ -413,6 +413,13 @@ void WiredTigerKVEngine::syncSizeInfo(bool sync) const {
         _sizeStorer->syncCache(sync);
     } catch (const WriteConflictException&) {
         // ignore, we'll try again later.
+    } catch (const UserException& ex) {
+        // re-throw exception if it's not WT_CACHE_FULL.
+        if (!_durable && ex.getCode() == ErrorCodes::ExceededMemoryLimit) {
+            error() << "size storer failed to sync cache: " << ex.what();
```

Comment:
I would elaborate in the error message to clarify that this error can be safely ignored and no data is lost.